### PR TITLE
Add support for batch purchase

### PR DESCRIPTION
**Pull request by antho1404**
Add batch purchase feature to purchase multiple offers in a single transaction

on `@liteflow/core` use the `client.exchange.batchPurchase` function
on `@liteflow/react` use `useBatchPurchase` hook

### CODE REVIEW

File: packages/react/src/useBatchPurchase.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const onProgress = useCallback(
+    (state) => {
+      switch (state.type) {
+        case 'TRANSACTION_SIGNATURE': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          break
+        }
+        case 'TRANSACTION_PENDING': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          setTransactionHash(state.payload.txHash)
```
Comment (NicolasMahe) [inline packages/react/src/useBatchPurchase.ts:42]: Shouldn't the transaction hash be reset when changing state?

File: packages/react/src/useBatchPurchase.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          break
+        }
+        case 'OWNERSHIP': {
+          setActiveProcess(BatchPurchaseStep.OWNERSHIP)
+          break
+        }
+        case 'OFFER_VALIDITY': {
+          console.log('validating offer...')
```
Comment (NicolasMahe) [inline packages/react/src/useBatchPurchase.ts:50]: console.log here.
could add a new `BatchPurchaseStep` or reuse an existing one if it make sense

File: packages/react/src/useBatchPurchase.ts (hunk 1 above)
Comment (NicolasMahe) [reply packages/react/src/useBatchPurchase.ts:42]: I see that's the same logic as https://github.com/liteflow-labs/liteflow-js/blob/5ca797516fb96ad626d5461ee2869c977653756b/packages/react/src/useAcceptOffer.ts
Let's keep the same logic then 👍

File: packages/react/src/useBatchPurchase.ts (hunk 2 above)
Comment (NicolasMahe) [reply packages/react/src/useBatchPurchase.ts:50]: There is also a console.log in this other hook that I guess you start from: https://github.com/liteflow-labs/liteflow-js/blob/5ca797516fb96ad626d5461ee2869c977653756b/packages/react/src/useAcceptOffer.ts#L59

File: packages/react/src/useBatchPurchase.ts (hunk 1 above)
Comment (antho1404) [reply packages/react/src/useBatchPurchase.ts:42]: This is following the same flow we have on other functions. It is definitely something that could be done, but it should be done across the whole library for consistency. I don't want to introduce inconsistency here.

Comment (NicolasMahe) [reply packages/react/src/useBatchPurchase.ts:42]: Actually maybe the reset should be done in the beginning of the `batchPurchase` function. It reset in the finally, but not on start.

File: packages/react/src/useBatchPurchase.ts (hunk 2 above)
Comment (antho1404) [reply packages/react/src/useBatchPurchase.ts:50]: Similar here, that's good, but it should be done for the whole library to include this step, adding it here without the rest will make things inconsistent.

File: packages/react/src/useBatchPurchase.ts (hunk 1 above)
Comment (antho1404) [reply packages/react/src/useBatchPurchase.ts:42]: moved to a dedicated issue https://github.com/liteflow-labs/liteflow-js/issues/203

File: packages/react/src/useBatchPurchase.ts (hunk 2 above)
Comment (antho1404) [reply packages/react/src/useBatchPurchase.ts:50]: moved to a dedicated issue https://github.com/liteflow-labs/liteflow-js/issues/204

**Pull request merged**